### PR TITLE
Live post notifications 2

### PR DESCRIPTION
**Pull request by robbiecarlton**

### CODE REVIEW

File: src/actions/fetchPosts.js [hunk 1]
```diff
@@ -30,6 +30,21 @@ export function fetchPosts (opts) {
...
   payload.path += '?' + querystring
 
-  let cache = {id: cacheId, bucket: 'postsByQuery', limit, offset, array: true}
-  return {type: FETCH_POSTS, payload, meta: {cache}}
+  var meta = {cache: {id: cacheId, bucket: 'postsByQuery', limit, offset, array: true}}
+
+  return {type: FETCH_POSTS, payload, meta}
+}
+
+export function checkFreshness (subject, id, posts, query = {}) {
```
Comment (robbiecarlton) [inline src/actions/fetchPosts.js:38]: not sure if this belongs here. But they have a lot of overlap, so it made sense to put them together. Maybe just need to change the filename?

File: src/components/SharingDropdown.js [hunk 2]
```diff
@@ -11,7 +11,9 @@ let share = href => {
 
 export const SharingDropdown = props => {
   let { url, text, ...otherProps } = props
-  let origin = typeof window !== 'undefined' ? window.location.origin : host
+  // let origin = typeof window !== 'undefined' ? window.location.origin : host
```
Comment (robbiecarlton) [inline src/components/SharingDropdown.js]: this is a temporary hack to get rid of a warning, not part of the final PR

**Comment by robbiecarlton** [comment]
@levity - how's this approach? modulo some tests. Do I have the thumbs up to generalize it to the other buckets?

**Comment by robbiecarlton** [comment]
screen shot 2016-03-03 at 12 13 02 pm
screen shot 2016-03-03 at 12 13 20 pm

Could maybe use a little cosmetic adjustment, but it does what it's supposed to

**Comment by robbiecarlton** [comment]
@levity good to merge?

File: css/index.less [hunk 3]
```diff
@@ -1483,6 +1483,30 @@ ul.tags {
   }
 }
 
+<<<<<<< HEAD
```
Comment (levity) [inline css/index.less]: ??

File: src/components/LiveStatusPoller.js [hunk 4]
```diff
@@ -11,10 +11,12 @@ export default class LiveStatusPoller extends React.Component {
   }
 
   componentDidMount () {
+    /*
```
Comment (levity) [inline src/components/LiveStatusPoller.js]: un-disable this?

File: src/components/RefreshButton.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  componentWillUnmount () {
+    window.removeEventListener('scroll', this.handleScrollEvents)
+  }
+
+  render () {
+    let { refresh } = this.props
+    let { isStatic } = this.state
+    return <span ref='placeholder'>
+      <VelocityTransitionGroup enter={{animation: 'slideDown'}} leave={{animation: 'slideUp'}}>
+        {refresh && <div className={cx('refresh-button', {'static': isStatic, 'floating': !isStatic})}
```
Comment (levity) [inline src/components/RefreshButton.js]: make the whole thing clickable

File: src/components/SharingDropdown.js (hunk 2 above)
Comment (levity) [inline src/components/SharingDropdown.js]: remove temporary hack?

File: src/reducers/index.js [hunk 6]
```diff
@@ -213,6 +217,18 @@ export default combineReducers({
     return state
   },
 
+  hasFreshPostsByQuery: keyedHasFreshItems(CHECK_FRESHNESS_POSTS, 'postsByQuery'),
+
+  totalPostsByQuery: (state = {}, action) => {
```
Comment (levity) [inline src/reducers/index.js]: looks like i missed refactoring this to use `keyedCounter`

File: test/util/caching.test.js [hunk 7]
```diff
@@ -30,7 +30,8 @@ describe('connectedListProps', () => {
     let expectedProps = {
```
Comment (levity) [inline test/util/caching.test.js:30]: more tests?

**Comment by robbiecarlton** [comment]
@levity - apologies, that last one was sloppy. How's this?

**Comment by levity** [comment]
:cool:

**Pull request merged**